src/filter.js

```diff
@@ -59,7 +67,16 @@ export class Filter extends CriteriaBuilder {
 
     //eslint-disable-next-line array-callback-return
     this.fieldElement.options.map(filter => {
-      this.fieldTypes[filter.name] = (filter.type === 'datetime') ? 'datetime-local' : filter.type;
+      this.fieldTypes[filter.value] = (filter.type === 'datetime') ? 'datetime-local' : filter.type;
+
+      if (filter.type === 'select') {
+        this.fieldEnumerations[filter.value] = (filter.options || []).map(option => {
+          return {
+            name : this.i18n.tr(option),
```

Comment:
No need for a hard dependency to `i18n`, simply pass the `name` on the `t` parameter in de view and as value.

```html
   <option repeat.for="field in fieldEnumerations" t="${field.name}">${field.name}</option
```